docs/src/model_creation/dsl_advanced.md

```diff
@@ -463,3 +463,77 @@ A reaction's metadata can be accessed using specific functions, e.g. `Catalyst.h
 rx = @reaction p, 0 --> X, [description="A production reaction"]
 Catalyst.getdescription(rx)
 ```
+
+## [Working with symbolic variables and the DSL](@id dsl_advanced_options_symbolics_and_DSL)
```

Comment:
Previously I moved these sections to the programmatic modelling section (as it has much better descriptions of symbolic variables and why this can be useful).  Then I planned to just put references to that section here.

I simply copied that section over here, it fits well here as well.